labgrid/driver/sshdriver.py

```diff
@@ -211,6 +281,27 @@ def _run(self, cmd, codec="utf-8", decodeerrors="strict", timeout=None): # pylin
             stderr.pop()
         return (stdout, stderr, sub.returncode)
 
+    @Driver.check_active
+    @step(args=['cmd'], result=True)
+    def popen(self, cmd: str):
+        if not self._check_keepalive():
+            raise ExecutionError("Keepalive no longer running")
+
+        complete_cmd = ["ssh", "-o", "LogLevel=QUIET", "-x", *self.ssh_prefix,
+                        "-p", str(self.networkservice.port), "-l", self.networkservice.username,
+                        self.networkservice.address, "-T", "--", '/bin/sh -c {}'.format(shlex.quote(cmd)),
+                        ]
+        self.logger.debug("Sending command: %s", complete_cmd)
```

Comment:
In the python standard library, popen doesn't use a shell by default. What is the motivation to do this here? Having the same interface as the `ShellDriver`?

Reply:
Yes, it should behave the same as ShellDriver; I think specifically you need to use a shell for redirection to work